tests/environment-edit.bats

```diff
@@ -0,0 +1,152 @@
+  mkdir -p "$PROJECT_DIR";
+  pushd "$PROJECT_DIR" >/dev/null || return;
+
+  "$FLOX_CLI" init
+  export MANIFEST_PATH="$PROJECT_DIR/.flox/test/pkgs/default/flox.nix";
+  echo "$ORIGINAL_MANIFEST_CONTENTS" > "$MANIFEST_PATH";
+  export EXTERNAL_MANIFEST_PATH="$PROJECT_DIR/input.nix";
+  echo "$NEW_MANIFEST_CONTENTS" > "$EXTERNAL_MANIFEST_PATH";
+}
+
+project_teardown() {
+  popd >/dev/null || return;
+  rm -rf "${PROJECT_DIR?}";
+  unset PROJECT_DIR;
+  unset ORIGINAL_MANIFEST_CONTENTS;
+  unset NEW_MANIFEST_CONTENTS;
+  unset MANIFEST_PATH;
+}
+
+# ---------------------------------------------------------------------------- #
+
+setup() {
+  common_test_setup;
+  project_setup;
+}
+teardown() {
+  project_teardown;
+  common_test_teardown;
+}
+
+setup_file() {
+  export FLOX_FEATURES_ENV=rust
+}
+
+# ---------------------------------------------------------------------------- #
+
+check_manifest_unchanged() {
+  current_contents=$(cat "$MANIFEST_PATH")
+  [[ "$current_contents" = "$ORIGINAL_MANIFEST_CONTENTS" ]]
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox edit' accepts contents via filename" {
+  run cat "$EXTERNAL_MANIFEST_PATH"
+  run "$FLOX_CLI" edit -f "$EXTERNAL_MANIFEST_PATH";
+  assert_success;
+  WRITTEN=$(cat "$MANIFEST_PATH");
+  assert_equal "$WRITTEN" "$NEW_MANIFEST_CONTENTS";
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox edit' accepts contents via pipe to stdin" {
+  run sh -c "cat ${EXTERNAL_MANIFEST_PATH} | ${FLOX_CLI} edit -f -";
+  assert_success;
+  # Get the contents as they appear in the actual manifest after the operation
+  WRITTEN=$(cat "$MANIFEST_PATH");
+  # Assert that it's the same as the contents we supplied
+  assert_equal "$WRITTEN" "$NEW_MANIFEST_CONTENTS";
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox edit' fails with invalid contents supplied via filename" {
+  echo "foo = " > "$EXTERNAL_MANIFEST_PATH";
+  run "$FLOX_CLI" edit -f "$EXTERNAL_MANIFEST_PATH";
+  assert_failure;
+  run check_manifest_unchanged;
+  assert_success;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox edit' fails with invalid contents supplied via stdin" {
+  run sh -c "echo 'foo = ;' | ${FLOX_CLI} edit -f -";
+  assert_failure;
+  run check_manifest_unchanged;
+  assert_success;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox edit' fails when provided filename doesn't exist" {
+  run "$FLOX_CLI" edit -f "does_not_exist.nix";
+  assert_failure;
+  run check_manifest_unchanged;
+  assert_success;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox edit' fails when EDITOR is not set" {
+  run "$FLOX_CLI" edit;
+  assert_failure;
+  run check_manifest_unchanged;
+  assert_success;
+}
```

Comment:
Would be good to test that EDITOR works with something like https://github.com/flox/flox/blob/51c602bcd00926c3604cac7204dd386078371951/tests/integration.bats#L191 and https://github.com/flox/flox/blob/51c602bcd00926c3604cac7204dd386078371951/tests/integration.bats#L166

Reply:
Not sure how that would all work with the interactivity, but it would also be nice to check the bad contents case with something like `EDITOR=./tests/make-invalid-edit`

Reply:
This feels like a pretty weird use of `EDITOR` in my opinion

Reply:
Writing these tests caught a bug we didn't notice. The current code doesn't invoke `EDITOR` against a temporary copy of the manifest, so a failed edit would still be left behind in the manifest. Easy fix though.